lib/dal/src/component/debug.rs

```diff
@@ -0,0 +1,177 @@
+use crate::workspace_snapshot::edge_weight::EdgeWeightKind;
+use crate::workspace_snapshot::node_weight::NodeWeightError;
+use crate::workspace_snapshot::WorkspaceSnapshotError;
+use crate::{
+    func::execution::FuncExecutionError, AttributeValue, AttributeValueId, Component, ComponentId,
+    DalContext, PropId, SchemaVariantId, SecretError, SecretId,
+};
+use crate::{ComponentError, FuncError, SchemaVariantError};
+
+type ComponentDebugViewResult<T> = Result<T, ComponentDebugViewError>;
+
+/// A generated view for an [`Component`](crate::Component) that contains metadata about each of
+/// the components attributes. Used for constructing a debug view of the component and also for
+/// cloning a component
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ComponentDebugView {
+    pub name: String,
+    pub schema_variant_id: SchemaVariantId,
+    pub attribute_tree: HashMap<AttributeValueId, Vec<AttributeValueId>>,
+    // pub input_sockets: Vec<AttributeDebugView>,
+    // pub output_sockets: Vec<AttributeDebugView>,
+}
+
+#[remain::sorted]
+#[derive(Error, Debug)]
+pub enum ComponentDebugViewError {
+    #[error(transparent)]
+    AttributeDebugViewError(#[from] AttributeDebugViewError),
+    #[error(transparent)]
+    AttributeValue(#[from] AttributeValueError),
+    #[error("Attribute Value tree badly constructed with root prop of {0}")]
+    AttributeValueTreeBad(AttributeValueId),
+    #[error("component error: {0}")]
+    Component(String),
+    #[error(transparent)]
+    ComponentError(#[from] ComponentError),
+    #[error(transparent)]
+    Func(#[from] FuncError),
+    #[error(transparent)]
+    FuncExecution(#[from] FuncExecutionError),
+    #[error("json pointer not found: {1:?} at {0}")]
+    JSONPointerNotFound(serde_json::Value, String),
+    #[error(transparent)]
+    NodeWeightError(#[from] NodeWeightError),
+    #[error("no internal provider for prop {0}")]
+    NoInternalProvider(PropId),
+    #[error("no root prop found for schema variant {0}")]
+    NoRootProp(SchemaVariantId),
+    #[error("schema variant not found for component {0}")]
+    NoSchemaVariant(ComponentId),
+    #[error("component not found {0}")]
+    NotFound(ComponentId),
+    #[error(transparent)]
+    Prop(#[from] PropError),
+    #[error(transparent)]
+    SchemaVariant(#[from] SchemaVariantError),
+    #[error(transparent)]
+    Secret(#[from] SecretError),
+    #[error("secret not found: {0}")]
+    SecretNotFound(SecretId),
+    #[error(transparent)]
+    SerdeJson(#[from] serde_json::Error),
+    #[error(transparent)]
+    UlidDecode(#[from] ulid::DecodeError),
+    #[error(transparent)]
+    WorkspaceSnapshotError(#[from] WorkspaceSnapshotError),
+}
+
+impl ComponentDebugView {
+    #[instrument(level = "info", skip_all)]
+    pub async fn new(ctx: &DalContext, component: &Component) -> ComponentDebugViewResult<Self> {
+        let schema_variant = Component::schema_variant(component, ctx).await?;
```

Comment:
nit: You don't need the whole schema variant. We may be able to just fetch the id from the node.